### ISSUE TITLE
An error that occurs when setting hyperparameters "'x' must be numeric."

### ISSUE THREAD

**Issue by akhrt01** [facebookexperimental/Robyn#834]
Let me ask you about an error that occurs when setting hyperparameters in Robyn, "'x' must be numeric.".
(I reinstalled Robyn today, so I think the version is 3.10.4.
 This is an error that did not occur when I used version 3.7.1.)

After entering the hyperparameters as below,

////////
hyperparameters <- list(
  media_A_alphas = c(0.5, 1),
  media_A_gammas = c(0.3, 1),
  media_A_thetas = c(0, 0.05),
  media_B_alphas = c(0.5, 1),
  media_B_gammas = c(0.3, 1),
  media_B_thetas = c(0, 0.2),
  train_size = c(0.5, 0.8)
)
////////

run "InputCollect <- robyn_inputs(InputCollect = InputCollect, hyperparameters = hyperparameters)".
Then the following error occurs.

////////
Error in `mutate()`:
ℹ In argument: `ds = cut(.data$ds, intervalType)`.
Caused by error in `cut.default()`:
!  'x' must be numeric.
---
Backtrace:
     ▆
  1. ├─Robyn::robyn_inputs(InputCollect = InputCollect, hyperparameters = hyperparameters)
  2. │ └─Robyn:::robyn_engineering(InputCollect, ...)
  3. │   └─Robyn:::prophet_decomp(...)
  4. │     └─Robyn:::set_holidays(dt_transform, dt_holidays, intervalType)
  5. │       └─... %>% ...
  6. ├─dplyr::summarise(...)
  7. ├─dplyr::group_by(., .data$ds, .data$country, .data$year)
  8. ├─dplyr::select(., .data$ds, .data$holiday, .data$country, .data$year)
  9. ├─dplyr::mutate(., ds = cut(.data$ds, intervalType))
 10. ├─dplyr:::mutate.data.frame(., ds = cut(.data$ds, intervalType))
 11. │ └─dplyr:::mutate_cols(.data, dplyr_quosures(...), by)
 12. │   ├─base::withCallingHandlers(...)
 13. │   └─dplyr:::mutate_col(dots[[i]], data, mask, new_columns)
 14. │     └─mask$eval_all_mutate(quo)
 15. │       └─dplyr (local) eval()
 16. ├─base::cut(.data$ds, intervalType)
 17. ├─base::cut.default(.data$ds, intervalType)
 18. │ └─base::stop("'x' must be numeric")
 19. └─base::.handleSimpleError(...)
 20.   └─dplyr (local) h(simpleError(msg, call))
 21.     └─rlang::abort(message, class = error_class, parent = parent, call = error_call)
////////

And run "print(InputCollect)", Robyn says"Hyper-parameters: Not set yet".
All data is either integer or numeric.
And I am converting a date column to date data as below.
////////
data$DATE <- as.IDate(data$DATE)
////////

What is the problem with this error and how can I resolve it?

**Comment by gufengzhou** [comment]
Your hyperparameter names don't look right. They should be media_A_alpha, _gamma, _theta etc. please read the demo.R guides

**Comment by gufengzhou** [comment]
We've actually deprecated data.table from Robyn completely. You're installing from CRAN right? You should get rid of this problem if you install from GitHub.

**Comment by gufengzhou** [comment]
I strongly recommend to use the latest version for future modelling. Backwards compatibility is not always guaranteed.

**Comment by gufengzhou** [comment]
Please reopen if necessary.
